<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CodeGenOptionsPage</class>
 <widget class="QWidget" name="CodeGenOptionsPage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>500</width>
    <height>539</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Code Generation Options</string>
  </property>
  <layout class="QVBoxLayout">
   <item>
    <widget class="QTabWidget" name="ui_tabWidgetMain">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="currentIndex">
      <number>1</number>
     </property>
     <widget class="QWidget" name="ui_tabGeneral">
      <attribute name="title">
       <string>General</string>
      </attribute>
      <layout class="QVBoxLayout">
       <item>
        <layout class="QVBoxLayout">
         <item>
          <widget class="QGroupBox" name="ui_groupBoxLanguage">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="title">
            <string>Language</string>
           </property>
           <layout class="QVBoxLayout">
            <item>
             <widget class="KComboBox" name="ui_SelectLanguageBox"/>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="ui_groupBoxFolders">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="title">
            <string>Folders</string>
           </property>
           <layout class="QVBoxLayout">
            <item>
             <widget class="QLabel" name="ui_textLabelFolder">
              <property name="text">
               <string>Write all generated files to folder:</string>
              </property>
              <property name="wordWrap">
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout">
              <item>
               <widget class="QLineEdit" name="ui_outputDir">
                <property name="whatsThis">
                 <string>Files generated by Code Generator will be written to this folder</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="ui_browseOutput">
                <property name="text">
                 <string>Bro&amp;wse...</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QCheckBox" name="ui_includeHeadings">
              <property name="text">
               <string>&amp;Include heading files from folder:</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout">
              <item>
               <widget class="QLineEdit" name="ui_headingsDir">
                <property name="whatsThis">
                 <string>Files in this folder will be used as heading files in the generated code</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="ui_browseHeadings">
                <property name="text">
                 <string>B&amp;rowse...</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="ui_groupBoxOverwritePolicy">
           <property name="title">
            <string>Overwrite Policy</string>
           </property>
           <layout class="QVBoxLayout">
            <item>
             <widget class="QLabel" name="ui_textLabelOverwritePolicy">
              <property name="text">
               <string>If a file with the same name as the name code 
generator wants to use as output file already exists:</string>
              </property>
              <property name="wordWrap">
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QVBoxLayout">
              <property name="spacing">
               <number>0</number>
              </property>
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>4</number>
              </property>
              <property name="bottomMargin">
               <number>4</number>
              </property>
              <item>
               <widget class="QRadioButton" name="ui_radioButtonOverwrite">
                <property name="whatsThis">
                 <string>Overwrite existing files if they exist in the destination folder</string>
                </property>
                <property name="text">
                 <string>O&amp;verwrite</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QRadioButton" name="ui_radioButtonAsk">
                <property name="whatsThis">
                 <string>If a file with the same name already exists, ask what to do</string>
                </property>
                <property name="text">
                 <string>As&amp;k</string>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QRadioButton" name="ui_radioButtonChangeName">
                <property name="whatsThis">
                 <string>If a file already exists in the destination folder, select a different name to use by adding a suffix to the file name</string>
                </property>
                <property name="text">
                 <string>&amp;Use a different name</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="ui_tabFormatting">
      <attribute name="title">
       <string>Formatting</string>
      </attribute>
      <layout class="QGridLayout">
       <item row="0" column="0">
        <layout class="QGridLayout">
         <item row="0" column="0">
          <widget class="QGroupBox" name="ui_groupBoxComment">
           <property name="title">
            <string>Comment Verbosity</string>
           </property>
           <layout class="QVBoxLayout">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout">
              <item>
               <spacer name="verticalSpacer">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>40</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QCheckBox" name="ui_forceDoc">
                <property name="whatsThis">
                 <string>Write comments &amp;for class and method documentation even if empty</string>
                </property>
                <property name="text">
                 <string>&amp;Write documentation comments even if empty</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="verticalSpacer_2">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>40</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_2">
                <item>
                 <widget class="QLabel" name="ui_textLabelSectionComments">
                  <property name="text">
                   <string>Write section comments:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QComboBox" name="ui_writeSectionComments">
                  <property name="whatsThis">
                   <string>Writes comments to indicate the different sections (public, private etc) in a class, even if the sections are empty</string>
                  </property>
                  <property name="currentIndex">
                   <number>0</number>
                  </property>
                  <item>
                   <property name="text">
                    <string>None</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>When section is non empty</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>Always</string>
                   </property>
                  </item>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <spacer name="verticalSpacer_3">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>40</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QGroupBox" name="ui_groupBoxLines">
           <property name="title">
            <string>Lines</string>
           </property>
           <layout class="QVBoxLayout">
            <item>
             <layout class="QHBoxLayout">
              <item>
               <widget class="QLabel" name="ui_textLabelType">
                <property name="text">
                 <string>Indentation type:</string>
                </property>
                <property name="wordWrap">
                 <bool>false</bool>
                </property>
               </widget>
              </item>
              <item>
               <widget class="KComboBox" name="ui_SelectIndentationTypeBox">
                <item>
                 <property name="text">
                  <string>No Indentation</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Tab</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Space</string>
                 </property>
                </item>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout">
              <item>
               <widget class="QLabel" name="ui_textLabelAmount">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Indentation amount:</string>
                </property>
                <property name="wordWrap">
                 <bool>false</bool>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="ui_SelectIndentationNumber"/>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout">
              <item>
               <widget class="QLabel" name="ui_textLabelStyle">
                <property name="text">
                 <string>Line ending style:</string>
                </property>
                <property name="wordWrap">
                 <bool>false</bool>
                </property>
               </widget>
              </item>
              <item>
               <widget class="KComboBox" name="ui_SelectEndLineCharsBox">
                <item>
                 <property name="text">
                  <string>*NIX (&quot;\n&quot;)</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Windows (&quot;\r\n&quot;)</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Mac (&quot;\r&quot;)</string>
                 </property>
                </item>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KComboBox</class>
   <extends>QComboBox</extends>
   <header>kcombobox.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
